### PR TITLE
Fall back on requests module when urllib fails to load a URL

### PR DESCRIPTION
**Pull request by dschuff**
This can happen on systems where OpenSSL is too old.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -926,11 +926,23 @@ def retrieve():
...
-        from urllib2 import urlopen
-      f = urlopen(url)
-      data = f.read()
+        from urllib2 import urlopen, URLError
+        try:
+          f = urlopen(url)
+          data = f.read()
+        except URLError as e:
+          try:
+            import requests
```
Comment (kripken) [inline tools/system_libs.py]: may be worth a comment that `requests` is used as a workaround for an older python?

**Comment by dschuff** [comment]
@quantum5 Is `requests` in py3k by default? If so, then that seems like a good idea. Otherwise I disagree that it should be the 'recommended' way to do things :)

**Comment by quantum5** [comment]
Unfortunately, `requests` is not part of the standard library. It is a rather popular library though and much more pleasant to use than `urllib`. I would personally use it first and use `urllib` as a fallback, but either way works.

**Comment by sbc100** [comment]
The problem is that we don't have any way today to use modules not part the standard library because we don't yet have any way to run any "install" or "post check-out" step before emscripten is expected to be functionally.  For the same reason we don't have any git sub-modules yes.   Once we improve the emsdk emscripten packaging story I think we can introduce some kind of "post-sync" step that git developers are expected to run, and then have this performes as part of the emsdk packaging (perhaps included in the emscripten release tar balls).

**Comment by quantum5** [comment]
Maybe an alternative to consider is to use `setup.py`, so then you can `pip install -e .` to install emscripten locally, as well as have it install dependencies. It would also put all the commands into `PATH`. Perhaps one day the end user can do `pip install emscripten`.

**Comment by sbc100** [comment]
Sure.   From my POV, it not so much about the specific command that needs to run, more about how to enforce that developers run that command to stay up-to-date.   Emscripten is a strange mix since we are have both python and node module dependencies in the same tree.  The npm people want to be able to to do `npm install emscripten` too :)   See #5774.

Node developers are used to running `npm install` after a `git pull`, python uses might run `pip install requirements.txt` or `setup.py`.. and everyone needs to run `git submodule-update`.   I guess we would want a single script to do any/all of this and some way to warn developers who forget to run it?

**Comment by sbc100** [comment]
I'm a little worried that this means we need to run out tests on both with and without the requests module.  

A recent bug that does use the requests module:  #11283

I this still needed?  I don't see the original bug that triggered this change?  I wonder if we can revert it now.   Having all those different methods seems like a bad idea.

**Pull request merged**